components/browser/search/src/test/java/mozilla/components/browser/search/SearchEngineManagerTest.kt

```diff
@@ -0,0 +1,147 @@
+                    mockSearchEngine("bing")))
+
+            val manager = SearchEngineManager(listOf(provider))
+
+            val engines = manager.getSearchEngines(RuntimeEnvironment.application)
+            assertEquals(3, engines.size)
+        }
+    }
+
+    @Test
+    fun `manager returns first engine if default cannot be found`() {
+        runBlocking {
+            val provider = mockProvider(listOf(
+                    mockSearchEngine("mozsearch"),
+                    mockSearchEngine("google"),
+                    mockSearchEngine("bing")))
+
+            val manager = SearchEngineManager(listOf(provider))
+
+            val default = manager.getDefaultSearchEngine(RuntimeEnvironment.application, "banana")
+            assertEquals("mozsearch", default.identifier)
+        }
+    }
+
+    @Test
+    fun `manager returns default engine with identifier if it exists`() {
+        runBlocking {
+            val provider = mockProvider(listOf(
+                    mockSearchEngine("mozsearch"),
+                    mockSearchEngine("google"),
+                    mockSearchEngine("bing")))
+
+            val manager = SearchEngineManager(listOf(provider))
+
+            val default = manager.getDefaultSearchEngine(RuntimeEnvironment.application, "bing")
+            assertEquals("bing", default.identifier)
+        }
+    }
+
+    @Test
+    fun `locale update broadcast will trigger reload`() {
+        runBlocking {
+            val provider = spy(mockProvider(listOf(
+                    mockSearchEngine("mozsearch"),
+                    mockSearchEngine("google"),
+                    mockSearchEngine("bing"))))
+
+            val manager = SearchEngineManager(listOf(provider))
+
+            val shadow = ShadowApplication.getInstance()
+            shadow.assertNoBroadcastListenersOfActionRegistered(
+                    RuntimeEnvironment.application,
+                    Intent.ACTION_LOCALE_CHANGED)
+
+            manager.registerForLocaleUpdates(RuntimeEnvironment.application)
+
+            assertTrue(shadow.registeredReceivers.find {
+                it.intentFilter.hasAction(Intent.ACTION_LOCALE_CHANGED)
+            } != null)
+
+            verify(provider, never()).loadSearchEngines(RuntimeEnvironment.application)
+
+            shadow.sendBroadcast(Intent(Intent.ACTION_LOCALE_CHANGED))
+
+            verify(provider).loadSearchEngines(RuntimeEnvironment.application)
+            verifyNoMoreInteractions(provider)
+        }
+    }
+
+    private fun List<SearchEngine>.assertContainsIdentifier(identifier: String) {
+        if (find { it.identifier == identifier } == null) {
+            throw AssertionError("$identifier not in list")
+        }
+    }
+
+    private fun mockProvider(engines: List<SearchEngine>): SearchEngineProvider =
+            object : SearchEngineProvider {
+                override suspend fun loadSearchEngines(context: Context): List<SearchEngine> {
```

Comment:
\[ktlint\] WARNING: \[modifier\-order\] Incorrect modifier order \(should be "suspend override"\) in column 17